File: BlogModule.Application/CommandHandlers/BlogPostComment/CreateBlogPostComment/CreateBlogPostCommentCommandHandler.cs
using System.ComponentModel.DataAnnotations;
using BlogModule.Application.Contract.BlogPostComments.CreateBlogPostComment;
using BlogModule.Domain.Models;
using BlogModule.Domain.Models.Ids;
using BlogModule.Domain.RepositoryInterfaces.BlogPostComment;
using Results.Domain;
using Shared.CQRS;
using Shared.CQRS.CommandHandlers;

namespace BlogModule.Application.CommandHandlers.BlogPostComment.CreateBlogPostComment;

public class CreateBlogPostCommentCommandHandler : ICommandHandler<CreateBlogPostCommentCommand, HandlerResult<CreateBlogPostCommentResponse, IErrorResult>>
{
    private readonly IBlogPostCommentCommandRepository _repository;

    public CreateBlogPostCommentCommandHandler(IBlogPostCommentCommandRepository repository)
    {
        _repository = repository;
    }

    public async Task<HandlerResult<CreateBlogPostCommentResponse,IErrorResult>> Handle(CreateBlogPostCommentCommand request, CancellationToken cancellationToken)
    {//dodać walidacje todo
        var context = new ValidationContext(request);
        Validator.ValidateObject(request, context, validateAllProperties: true);
        
        var newComment = new BlogPostCommentModel
        {
            Id = new BlogPostCommentId(Guid.NewGuid()),
            BlogPostId = request.BlogPostId,
            UserId = request.UserId,
            Description = request.Description,
            DateAdded = DateTime.Now
        };

        await _repository.AddAsync(newComment, cancellationToken);

        var response = new CreateBlogPostCommentResponse
        {
            Id = newComment.Id,
            Title = "Comment Created",
            Message = "The blog post comment has been successfully created."
        };

        return response;
    }
}

File: BlogApi/Application/command_handlers/blog_post_comment/create_blog_post_comment/create_blog_post_comment_command_handler.go
package commands

import (
	contract "blog-api/Application.contract/BlogPostComments/CreateBlogPostComment"
	models "blog-api/Domain/models/domestic"
	repository_interfaces "blog-api/Domain/repository_interfaces/blog_post_comment_repository"
	blog_repository "blog-api/Domain/repository_interfaces/blog_post_repository"
	"blog-api/Domain/responses"
	data_fetcher "blog-api/Domain/service_interfaces"
	"context"
)

type CreateBlogPostCommentCommandHandler struct {
	blogPostCommentCommandRepository repository_interfaces.BlogPostCommentCommandRepository
	blogPostQueryRepository          blog_repository.BlogPostQueryRepository
	dataFetcher                      data_fetcher.DataFetcher
}

func NewCreateBlogPostCommentCommandHandler(
	blogPostCommentCommandRepository repository_interfaces.BlogPostCommentCommandRepository,
	blogPostQueryRepository blog_repository.BlogPostQueryRepository,
	dataFetcher data_fetcher.DataFetcher,
) *CreateBlogPostCommentCommandHandler {
	return &CreateBlogPostCommentCommandHandler{
		blogPostCommentCommandRepository: blogPostCommentCommandRepository,
		blogPostQueryRepository:          blogPostQueryRepository,
		dataFetcher:                      dataFetcher,
	}
}

func (h *CreateBlogPostCommentCommandHandler) Handle(ctx context.Context, command *CreateBlogPostCommentCommand) (*contract.CreateBlogPostCommentResponse, error) {
	userInfo, err := h.dataFetcher.GetUserInternalInfo(command.JwtToken)
	if err != nil {
		response := responses.NewResponse[contract.CreateBlogPostCommentResponse](401, "Unauthorized")
		return &response, nil
	}

	blogPost, err := h.blogPostQueryRepository.GetBlogPostByID(command.BlogPostId)
	if err != nil || blogPost == nil {
		response := responses.NewResponse[contract.CreateBlogPostCommentResponse](404, "Blog post not found")
		return &response, nil
	}

	comment := &models.BlogPostCommentModel{
		Description: command.Description,
		BlogPostId:  command.BlogPostId,
		UserId:      userInfo.ID,
	}

	commentID, err := h.blogPostCommentCommandRepository.AddComment(ctx, comment)
	if err != nil {
		response := responses.NewResponse[contract.CreateBlogPostCommentResponse](500, "Failed to create comment")
		return &response, nil
	}

	return &contract.CreateBlogPostCommentResponse{
		BaseResponse: responses.NewBaseResponse(200, "Comment created successfully"),
		Id:           commentID,
	}, nil
}
